src/topotoolbox/grid_object.py

```diff
@@ -302,6 +304,78 @@ def excesstopography(
 
         return result
 
+    def filter(self, method: str = 'mean', kernelsize: int = 3) -> 'GridObject':
+        """The function filter is a wrapper around various image filtering
+        algorithms. Only filters with rectangular kernels of uneven
+        dimensions are supported.
+
+        Parameters
+        ----------
+        method : str, optional
+            Which method will be used to filter the DEM: ['mean', 'average',
+            'median', 'sobel', 'scharr', 'wiener', 'std'], by default 'mean'
+        kernelsize : int, optional
+            Size of the kernel that will be applied. Note that ['sobel',
+            'scharr'] require that the kernelsize is 3, by default 3
+
+        Returns
+        -------
+        GridObject
+            The filtered DEM as a GridObject.
+
+        Raises
+        ------
+        ValueError
+            If the kernelsize does not match the requirements of this function
+            or the selected method is not implemented in the function.
+        """
+
+        valid_methods = ['mean', 'average', 'median',
+                         'sobel', 'scharr', 'wiener', 'std']
+
+        if method in ['mean', 'average']:
+            kernel = np.ones((kernelsize, kernelsize)) / kernelsize**2
+            filtered = convolve(self.z, kernel, mode='nearest')
+
+        elif method in ['median']:
+            filtered = median_filter(self.z, size=kernelsize, mode='reflect')
+
+        elif method in ['sobel', 'scharr']:
+            if kernelsize != 3:
+                arr = f"The method '{method}' only works with a 3x3 kernel'."
+                raise ValueError(arr) from None
+
+            if method == 'sobel':
+                ky = np.array([[1, 2, 1], [0, 0, 0], [-1, -2, -1]])
```

Comment:
There is also a `'sobel` function in `scipy.ndimage`, but it seems to give the same results as the explicit filter you use here, and I don't think there is any real benefit to using one over the other.